### DISCUSSION TITLE
Infererd type of Data is wrong with the axios-typescript example

### DISCUSSION THREAD

**Discussion by svenlombaert** [vercel/swr#2617] [Q&A] [answered]
# Bug report

## Description / Observed Behavior

I think the returned data property from the axios typescript example should be a union type of Data | undefined. It's always Data, even with suspense set to false

## Expected Behavior

Union type of Data | undefined is returned

## Repro Steps / Code Example

https://codesandbox.io/s/swr-axios-ts-example-bp4w6s?file=/src/index.tsx

![Screenshot 2023-05-12 at 14 31 47](https://github.com/vercel/swr/assets/2846986/02552dc5-ac77-4997-9c63-3024151666ec)

## Additional Context

SWR version: 2.1.5

Edit: with the exact same versions of axios, typescript and swr in a react-native project, the type *is* inferred correctly

**Comment by huozhi** [comment, marked as answer]
It seems working properly with next.js locally, could be sth wrong with the reproduction on codesandbox? I'll close this issue for now as it's not a problem of the library. Can carry on in the discussion

**Reply by svenlombaert** [reply]
Hmm, I literally copy-pasted from the example in this repository though. To be honest, I'm actually puzzled myself since the exact same code does work in a different project and it feel like I'm delusional 😅

**Reply by svenlombaert** [reply]
Someone turned of 'strict' in our tsconfig - all good now 🤦
